### PR TITLE
Split RTPStats into receiver and sender.

### PR DESCRIPTION
**Pull request by boks1971**
For receiver, short types are input and need to calculate extended type.

For sender (subscriber), it can operate only in extended type. This makes the subscriber side a little simpler and should make it more efficient as it can do simple comparisons in extended type space.

There was also an issue with subscriber using shorter type and calculating extended type. When subscriber starts after the publisher has already rolled over in sequence number OR timestamp, when subsequent publisher side sender reports are used to adjust subscriber time stamps, they were out of whack. Using extended type on subscriber does not face that.

@davidzhao @cnderrauber @paulwe - This is pretty much splitting up `RTPStats` into receive side and send side. And trying to keep the common bits in `base`. Some of the stats update in `Update` method of receiver/sender can be pushed to `base`, but it got too cumbersome passing in bunch of stuff and also messing with control flow. So, it is repeated for now. Will check if I can combine that better.

### CODE REVIEW

File: pkg/sfu/connectionquality/connectionstats.go [hunk 1]
```diff
@@ -40,7 +40,7 @@ type ConnectionStatsParams struct {
 	IncludeRTT                bool
 	IncludeJitter             bool
 	GetDeltaStats             func() map[uint32]*buffer.StreamStatsWithLayers
-	GetDeltaStatsOverridden   func() map[uint32]*buffer.StreamStatsWithLayers
+	GetDeltaStatsSender       func() map[uint32]*buffer.StreamStatsWithLayers
 	GetLastReceiverReportTime func() time.Time
 	GetTotalPacketsSent       func() uint64
```
Comment (paulwe) [inline pkg/sfu/connectionquality/connectionstats.go:45]: would an interface or maybe 2 make sense here?

**Review by paulwe — APPROVED** [review]
mega refactor ✨ lgtm!

File: pkg/sfu/connectionquality/connectionstats.go (hunk 1 above)
Comment (boks1971) [reply pkg/sfu/connectionquality/connectionstats.go:45]: Yeah will look at that Paul. It is a bit whacky based on publisher or subscribers using it, but can dummy impl methods to satisfy the interface. Will make a separate PR for that.

**Review by davidzhao — APPROVED** [review]
lgtm!

**Review by cnderrauber — APPROVED** [review]
Great work to improve the stats!

**Pull request merged**